### PR TITLE
[aptos-stdlib] Any type

### PR DESCRIPTION
**Pull request by wrwg**
### Description

This adds the `Any` type which allows polymorphic representation of values.

### Test Plan

Added test

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3291)

### CODE REVIEW

**Review by areshand — APPROVED** [review]

File: aptos-move/framework/aptos-stdlib/sources/any.move [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /// changes one can do
+    ///
+    /// ```move
+    ///   struct Resource {
+    ///      field: Type,
+    ///      ...
+    ///      extension: Option<Any>
+    ///   }
+    /// ```
+    struct Any has drop, store {
```
Comment (vgao1996) [inline aptos-move/framework/aptos-stdlib/sources/any.move:21]: I wonder if we need multiple flavors of Any with different abilities

File: aptos-move/framework/aptos-framework/sources/util.move [hunk 2]
```diff
@@ -4,7 +4,9 @@ module aptos_framework::util {
     friend aptos_framework::gas_schedule;
 
     /// Native function to deserialize a type T.
-    /// TODO: may want to move it in extra module if needed also in other places inside of the Fx.
-    /// However, should not make this function public outside of the Fx.
-    public(friend) native fun from_bytes<T: copy + drop>(bytes: vector<u8>): T;
+    ///
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/util.move:7]: this probably needs to move to aptos-stdlib even if it ... requires friends

File: aptos-move/framework/aptos-stdlib/sources/any.move (hunk 1 above)
Comment (wrwg) [reply aptos-move/framework/aptos-stdlib/sources/any.move:21]: Maybe, but lets start with this one.

**Review by davidiw — APPROVED** [review]
looks good otherwise

**Comment by davidiw** [comment]
I'd be curious the perf impact of using strings and byte conversions

File: aptos-move/framework/aptos-framework/sources/util.move (hunk 2 above)
Comment (wrwg) [reply aptos-move/framework/aptos-framework/sources/util.move:7]: No it is need here, both for code and gas

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/util.move:7]: Hmm weird to have it in both framework and aptos-stdlib. Can't we make the framework package depend on the aptos-stdlib and stdlib packages and have just one single from_bytes function in aptos-stdlib?

Comment (wrwg) [reply aptos-move/framework/aptos-framework/sources/util.move:7]: We can't. We don't want to make `from_bytes` public, and a friend can't be used across packages. This is also documented in the code.

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/util.move:7]: Hmm I see. Thanks for the clarification.

**Review by movekevin — APPROVED** [review]

**Pull request merged**